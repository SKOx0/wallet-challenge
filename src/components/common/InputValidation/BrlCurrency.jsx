import React from 'react';
import { Field } from 'redux-form';
import { Help } from 'bloomer';
import PropTypes from 'prop-types';
import BrlCurrencyInput from '../Input/BrlCurrency';

const validationField = ({
  input, label, type, meta: { touched, error }, readOnly, onChangeCurrency
}) => (
  <div>
    <BrlCurrencyInput {...input} type={type} placeholder={label} readOnly={readOnly} onChange={onChangeCurrency} />
    {touched && error && <Help isColor="danger">{error}</Help>}
  </div>
);

const BrlCurrency = ({ ...rest }) => (<Field component={validationField} {...rest} />);

export default BrlCurrency;

validationField.propTypes = {
  input: PropTypes.object,
  label: PropTypes.string,
  type: PropTypes.string,
  meta: PropTypes.object,
  readOnly: PropTypes.bool,
  onChangeCurrency: PropTypes.func
};
