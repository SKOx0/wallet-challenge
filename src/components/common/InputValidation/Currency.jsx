import React from 'react';
import { Field } from 'redux-form';
import { Help } from 'bloomer';
import PropTypes from 'prop-types';
import CurrencyInput from '../Input/Currency';

const validationField = ({
  input, label, type, meta: { touched, error }, readOnly, onChangeCurrency, currencyConfig, currency
}) => (
  <div>
    <CurrencyInput {...input} type={type} placeholder={label} readOnly={readOnly} onChange={onChangeCurrency} config={currencyConfig} currency={currency} />
    {touched && error && <Help isColor="danger">{error}</Help>}
  </div>
);

const BrlCurrency = ({ ...rest }) => (<Field component={validationField} {...rest} />);

export default BrlCurrency;

validationField.propTypes = {
  input: PropTypes.object,
  label: PropTypes.string,
  type: PropTypes.string,
  meta: PropTypes.object,
  readOnly: PropTypes.bool,
  onChangeCurrency: PropTypes.func,
  currencyConfig: PropTypes.object,
  currency: PropTypes.string
};
