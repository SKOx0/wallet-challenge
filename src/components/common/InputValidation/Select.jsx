import React from 'react';
import { Field } from 'redux-form';
import Input from '../Input';

const SelectValidation = (props) => (<Field component="select" {...props} />);

export default Input.withComponent(SelectValidation);
