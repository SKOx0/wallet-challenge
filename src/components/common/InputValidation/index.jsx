import React from 'react';
import { Field } from 'redux-form';
import { Help } from 'bloomer';
import PropTypes from 'prop-types';
import Input from '../Input';

const validationField = ({
  input, label, type, meta: { touched, error }
}) => (
  <div>
    <Input {...input} type={type} placeholder={label} />
    {touched && error && <Help isColor="danger">{error}</Help>}
  </div>
);

const InputValidation = (props) => (<Field component={validationField} {...props} />);

export default InputValidation;

validationField.propTypes = {
  input: PropTypes.object,
  label: PropTypes.string,
  type: PropTypes.string,
  meta: PropTypes.object
};
