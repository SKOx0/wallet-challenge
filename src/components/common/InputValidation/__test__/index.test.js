import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import InputValidation from '../index';

describe('<InputValidation />', () => {
  it('deve ser igual ao snapshot', () => {
    const renderedComponent = shallow(<InputValidation />);

    expect(toJson(renderedComponent)).toMatchSnapshot();
  });
});
