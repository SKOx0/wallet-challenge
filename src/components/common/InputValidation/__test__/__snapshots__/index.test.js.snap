// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputValidation /> deve ser igual ao snapshot 1`] = `
<Field
  component={[Function]}
/>
`;
